nodes/geometry/point_inside_camera_frustrum.py

```diff
@@ -0,0 +1,42 @@
+import bpy
+from bpy.props import *
+from ... base_types.node import AnimationNode
+import bpy_extras
+
+class PointInCameraFrustrum(bpy.types.Node, AnimationNode):
+    bl_idname = "an_PointInCameraFrustrum"
+    bl_label = "Point in Camera Frustrum"
+    
+    def create(self):
+        self.newInput("Scene", "Scene", "scene", hide = True)
+        self.newInput("Object", "Camera", "camera")
+        self.newInput("Vector", "3d Point", "point", value=(0, 0, 0))
+        self.newInput("Float", "Threshold % (positive = interior)", "threshold", value = 0.0)
+        
+        self.newOutput("Float", "Image u", "u")
+        self.newOutput("Float", "Image v", "v")
+        self.newOutput("Float", "Z depth", "z")
+        self.newOutput("Boolean", "Is in frustrum", "visible")
+        
+    def getExecutionCode(self):
+        yield "u, v, z, visible = self.isInFrustrum(scene, camera, point, threshold)"
+            
+    def getUsedModules(self):
+        return ["bpy_extras"]
+    
+    def isInFrustrum(self, scene, camera, point, threshold):
+        if camera is None:
+            return 0, 0, 0, False, 0
+        if camera.type != 'CAMERA':
```

Comment:
both if conditions can be combined into ``if getattr(camera, "type", "") != "CAMERA":``
This way we only need to write the default values once